src/jsx/components/results_page.jsx

```diff
@@ -38,19 +38,18 @@ class ResultsPage extends React.Component {
       self.setState({ json: self.props.data });
     }
 
-    // Decide if data is a URL or the results JSON
     if (typeof this.props.fasta == "string") {
-        try {
+      self.setState({ fastaPath: this.props.fasta }); // Set the fasta path for comparing on updates to see if it's new data.  
+      try{
+        d3.text(this.props.fasta, function(data) {
           let fasta_json = fastaParser(JSON.parse(data).fasta);

```

Comment:
Hmm, I think the discrepancy comes from the fact that in some methods, retrieval is done from the results.ejs template (the script executed by the page itself) -- https://github.com/veg/datamonkey-js/blob/master/app/templates/absrel/results.ejs#L10. 

Then you have BUSTED, which doesn't do that. https://github.com/veg/datamonkey-js/blob/master/app/templates/absrel/results.ejs#L10